### PR TITLE
Document ShutdownDelay

### LINKED ISSUE #32600: Document fix for 503's during app recycle in IIS
### Description

See
* https://github.com/dotnet/AspNetCore.Docs/issues/32361#issuecomment-2108856264
* https://github.com/dotnet/AspNetCore.Docs/pull/32598#issuecomment-2118220491

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/web-config?view=aspnetcore-8.0#configuration-of-aspnet-core-module-with-webconfig

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/iis/web-config.md

### Document ID

f47ff3cc-cd78-d41c-af1c-ef1e752cae88

### Article author

@Rick-Anderson

---
[Associated WorkItem - 256990](https://dev.azure.com/msft-skilling/Content/_workitems/edit/256990)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #32600 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/host-and-deploy/iis/advanced.md](https://github.com/dotnet/AspNetCore.Docs/blob/8e390133fe36aa40709bf265d758174d76147cc5/aspnetcore/host-and-deploy/iis/advanced.md) | [Advanced configuration of the ASP.NET Core Module and IIS](https://review.learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/advanced?branch=pr-en-us-32637) |

### CODE REVIEW

File: aspnetcore/host-and-deploy/iis/advanced.md [hunk 1]
```diff
@@ -48,6 +48,29 @@ The `disallowRotationOnConfigChange` setting is intended for blue/green scenario
 
 This setting corresponds to the API <xref:Microsoft.Web.Administration.ApplicationPoolRecycling.DisallowRotationOnConfigChange?displayProperty=nameWithType>
 
+## Fix for 503's during app recycle in IIS
```
Comment (BrennanConroy) [inline aspnetcore/host-and-deploy/iis/advanced.md:51]: Not a fan of the title, here's some other thoughts:

Reduce 503 likelihood during app recycle
Delay managed app shutdown during app recycle
Delay server shutdown on recycle

File: aspnetcore/host-and-deploy/iis/advanced.md [hunk 2]
```diff
@@ -48,6 +48,29 @@ The `disallowRotationOnConfigChange` setting is intended for blue/green scenario
...
+
+By default there is a one second delay between when IIS is notified of a recycle or shutdown and when ANCM tells the managed server to start shutting down. The delay is configurable via the `ANCM_shutdownDelay` environment variable or by setting the `shutdownDelay` handler setting. Both values are in milliseconds. The delay is mainly to reduce the likelihood of a race where:
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/iis/advanced.md:53]:
```suggestion
By default there is a one second delay between when IIS is notified of a recycle or shutdown and when ANCM tells the managed server to initiate shutdown. The delay is configurable via the `ANCM_shutdownDelay` environment variable or by setting the `shutdownDelay` handler setting. Both values are in milliseconds. The delay is primarily to reduce the likelihood of a race where:
```

File: aspnetcore/host-and-deploy/iis/advanced.md [hunk 3]
```diff
@@ -48,6 +48,29 @@ The `disallowRotationOnConfigChange` setting is intended for blue/green scenario
...
+
+* IIS hasn't started queuing requests to go to the new app.
+* ANCM starts rejecting new requests that come into the old app.
+
+This setting doesn't mean incoming requests will be delayed by this amount. It just means that the old app instance will start shutting down after this timeout occurs. Slower machines or machines with heavier CPU usage might need to adjust this value to reduce 503 likelihood.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/iis/advanced.md:58]:
```suggestion
This setting doesn't mean incoming requests will be delayed by this amount. The setting indicates that the old app instance will start shutting down after the timeout occurs. Slower machines or machines with heavier CPU usage might need to adjust this value to reduce 503 likelihood.
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/host-and-deploy/iis/advanced.md (hunk 1 above)
Comment (tdykstra) [inline aspnetcore/host-and-deploy/iis/advanced.md:51]:
```suggestion
## Reduce 503 likelihood during app recycle```

File: aspnetcore/host-and-deploy/iis/advanced.md [hunk 4]
```diff
@@ -48,6 +48,29 @@ The `disallowRotationOnConfigChange` setting is intended for blue/green scenario
 
 This setting corresponds to the API <xref:Microsoft.Web.Administration.ApplicationPoolRecycling.DisallowRotationOnConfigChange?displayProperty=nameWithType>
 
+## Reduce 503 likelihood during app recycle```
```
Comment (tdykstra) [inline aspnetcore/host-and-deploy/iis/advanced.md:51]:
```suggestion
## Reduce 503 likelihood during app recycle
```

**Review by Rick-Anderson — APPROVED** [review]

**Pull request merged**